### PR TITLE
Use TypedValueCompat

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Simplify pixel conversion code using `TypedValueCompat` methods.

### CODE REVIEW

**Comment by FloEdelmann** [comment]
Could you check whether the conversion functions in `ChangelogFragment` can also be simplified using these extension functions?

**Review by westnordost — APPROVED** [review]
LGTM

File: app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt [hunk 1]
```diff
@@ -14,28 +14,20 @@ import androidx.core.content.ContextCompat
...
 /** return the number of density independent pixels for the given pixels */
-fun Context.pxToDp(px: Int): Float = px / resources.displayMetrics.density
+fun Context.pxToDp(px: Int): Float = resources.pxToDp(px)
 
 /** return the number of pixels for the given density independent pixels */
-fun Context.dpToPx(dp: Int): Float = dp * resources.displayMetrics.density
-
-/** return the number of pixels for the given scalable pixels */
-fun Context.spToPx(sp: Int): Float = sp * resources.displayMetrics.scaledDensity
+fun Context.dpToPx(dp: Int): Float = resources.dpToPx(dp)
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt:30]: For consistency: Should we maybe move all those px/dp/sp-related extension functions from `Context` to `Resources` and use it there directly?

**Review by FloEdelmann — APPROVED** [review]
Looks good to me now. Thanks for incorporating all the feedback! Note that I haven't checked out the code locally, so it's only a code review.

**Comment by westnordost** [comment]
LGTM too. (Also before.)

Note that having shortcuts like `fun Context.a() = resources.a()` is completely fine by me. But I don't mind so much either way, especially since the whole Android view stuff is due to be replaced with Jetpack compose. (Currently looking into setting it up.)

**Pull request merged**